gaps/gap-4_comp_manifest/gap-4_comp_manifest.md

```diff
@@ -0,0 +1,278 @@
+---
+gap: 4
+title: Computation Manifest
+description:
+author: mf (@mfranciszkiewicz)
+status: Draft
+type: Feature
+---
+
+## Abstract
+
+This GAP introduces a manifest file specification and a set of GCRS (Golem
+Computation Resource Standards) properties (herein after referred to as
+"Computation Manifest"), which allow Requestors to declare the constraints of
+computations they may execute on a Provider's machine.
+
+## Motivation
+
+The need for a Computation Manifest originates from the upcoming Internet
+access feature, where each outbound request is routed via Provider's host
+operating system. In order to prevent unwanted / illegal behavior on Provider's
+behalf, she needs to remain in control of actions that Requestors can execute
+on her machine.
+
+With Computation Manifests, Requestors constrain themselves to
+a certain set of allowed actions, to be negotiated with and approved by a
+Provider. Requestors' actions will be verified against the Manifest during
+computation.
+
+## Specification
+
+Computation Manifest can take one of the following forms:
+
+- an embedded JSON / YAML object within an external configuration file,
+- `golem.srv.comp.manifest` namespace properties, an extension to
+  Golem Computation Resource Standards.
+
+The Computation Manifest is expected to be included in a Computation Payload
+Manifest ([GAP](https://github.com/golemfactory/golem-architecture/pull/28))
+file, where its hash is co-signed with Payload hashes. If a Computation
+Manifest is included in the Demand (and thus, Agreement), its role is to
+derive the properties from and override the initially sourced Manifest.
+
```

Comment:
Can we please also indicate the "lifecycle" of the Manifest - ie. 
- who creates it?
- what happens with it? (ie. if it is signed and then validated - then by whom, and against what?)
- where does the manifest reside (eg. is it part of the Golem application "package", together with payload GVMIs?)

Reply:
Done